api/millenium_patron.py

```diff
@@ -385,18 +408,27 @@ def patron_dump_to_patrondata(self, current_identifier, content):
         potential_identifiers = []
         for f, v in self._extract_text_nodes(content):
             k = self._code_from_field(f)
-            if k == self.BARCODE_FIELD:
+
+            # Check to see if the key should be treated as if it is a patron identifier.
+            # This is dependent on a configuration setting. The key will also have a
+            # chance to be treated as a different kind of field below (for example, if
+            # the configuration says that the username field 'pu' should be treated as
+            # a patron identifier, we _also_ want to treat is as a username below; both
```

Comment:
minor: typo "...treat [is] as a username..." -> "...treat [it] as a username..."